draft-ietf-quic-applicability.md

```diff
@@ -452,12 +452,20 @@ human-readable information to an error code. Note that QUIC RESET_STREAM and
 STOP_SENDING frames provide similar capablities. Usually application error codes
 are defined to be applicabile to all three frames.
 
-Alternatively, a QUIC connection will be silently closed by each endpoint
-separately after an idle timeout. The idle timeout is announce for each
-endpoint during connection established and should be accessible by the
-application. If an application desires to keep the connection open for longer
-than the announced timeout, it can send keep-alives messages. See
-{#resumption-v-keepalive} for further guidance.
+Alternatively, a QUIC connection can be silently closed by each endpoint
+separately after an idle timeout. If enabled as indcated by a transport
+parameter in the handshake, the idle timeout is announced for each
+endpoint during connection establishment and the effective value for this
+connection is the minimum of these two advertised value. An application
+therefore should be able to configure its own maximum value as well as
+have access to the computed minimum value for this connection. An application
```

Comment:
I don’t understand why an implementation needs access to the computed value. A reasonable QUIC implementation could take the application’s idle timeout and then take care of sending PINGs to the peer to keep the connection open.

Reply:
The text here, I think, is appropriate for general-purpose QUIC implementations (atop which multiple applications could be bound), but yeah, it could be either-or for applications tightly bound to their implementation, based on application keepalive logic.

Reply:
I think that - at the level of this document - this is fine advice.  Applications do benefit from knowing when a timeout is imminent because they might want to manage the risk of failure for transactions that can't be easily retried.  At or near the timeout is not a great time to start something.